3rd_party/3rd_party.cmake

```diff
@@ -1,274 +1,272 @@
+        foreach(_OLD_LIB ${_OLD_LIBS})
+          if(_OLD_LIB)
+            file(REMOVE ${_OLD_LIB})
+          endif()
+        endforeach()
+        foreach(_INST_LIB ${_INST_LIBS})
+          # Resolve symlinks so that we have a handle
+          # on a plain file.
+          file(REAL_PATH ${_INST_LIB} _RESOLVED_PATH)
+          # Strip off any leading directories
+          set(_RESOLVED_LIB)
+          if(${_RESOLVED_PATH} MATCHES "/([^/]+)$")
+            set(_RESOLVED_LIB ${CMAKE_MATCH_1})
+          endif()
+          if(${_RESOLVED_PATH} MATCHES "/([^/]+${_postfix})")
+            set(_LIB ${CMAKE_MATCH_1})
+            file(COPY ${_RESOLVED_PATH} DESTINATION ${INSTALL_DIR})
+            # The file we copied may not be precisely the name that we want as we had to resolve symlinks.
+            # If so we need to rename it.
+            if(NOT EXISTS ${INSTALL_DIR}/${_LIB})
+              file(RENAME ${INSTALL_DIR}/${_RESOLVED_LIB} ${INSTALL_DIR}/${_LIB})
+            endif()
+            file(CHMOD ${INSTALL_DIR}/${_LIB} PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
+          else()
+            file(COPY ${_RESOLVED_PATH} DESTINATION ${INSTALL_DIR})
+            file(CHMOD ${INSTALL_DIR}/${_RESOLVED_LIB} PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
+          endif()
+        endforeach()
+      endif()
+    endforeach()
+  else()
+    message(FATAL_ERROR "${_target} not found")
+    return()
+  endif()
+endfunction()
+
+function(third_party _arg)
+  if("${_arg}" STREQUAL "check")
+    set(INSTALL_DIR "")
+  elseif("${_arg}" STREQUAL "add")
+    if(${ARGC} LESS 2)
+      message(FATAL_ERROR "Install directory required for 'add'")
+      return()
+    endif()
+    message(STATUS "Setting install directory to ${ARGV1}")
+    set(INSTALL_DIR ${ARGV1})
+  else()
+    message(FATAL_ERROR "Action required: 'add' or 'check'")
+    return()
+  endif()
+
+  file(MAKE_DIRECTORY "${INSTALL_DIR}")
+
+  install_libs("Boost" ${BOOST_LOCATION} "boost" "${BOOST_COMPILER}*${BOOST_EXTENSION}" ${BOOST_LIBRARIES})
+  install_libs("libxml2" ${XML_LOCATION} "" "${XML_EXTENSION}" "libxml2")
+  install_libs("gcc runtime library" ${GCC_RT_LOCATION} "" "${GCC_RT_EXTENSION}" "libgcc_s")
+  install_libs("C++ standard library" ${STL_LOCATION} "" "${STL_EXTENSION}" "${STL_PATTERN}")
+  install_libs("OMP runtime library" ${OMP_LOCATION} "" "${OMP_EXTENSION}" "${OMP_PATTERN}")
+  install_libs("zlib" ${ZLIB_LOCATION} "" "${ZLIB_EXTENSION}" "zlib")
+  install_libs("Torch libraries" ${TORCH_LOCATION} "" "${TORCH_EXTENSION}" "${TORCH_LIBRARIES}")
+  install_libs("Intel MKL libraries" ${MKL_LOCATION} "${MKL_PREFIX}" "${MKL_EXTENSION}" "${MKL_LIBRARIES}")
+
+  # Special extra platform-specific processing
+  if ("${HOST_SYSTEM_NAME}" STREQUAL "linux")
+    if (INSTALL_DIR AND NOT DEFINED ENV{CPP_CROSS_COMPILE} OR NOT "$ENV{CPP_CROSS_COMPILE}" STREQUAL "macosx")
```

Comment:
I think given the code above these two conditions together could now be simplified to:

```
if (INSTALL_DIR AND CMAKE_SYSTEM_NAME STREQUAL "Linux")
```

Reply:
Thanks Dave. At the moment it seems to be working so I think I'll leave well enough alone!